### PR TITLE
hpilo_info: add some functionalities

### PR DESCRIPTION
**Pull request by Davidffry**
##### SUMMARY

Add some other infos to improve the retrieve

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

hpilo_info

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by russoz** [comment]
hi @Davidffry Thank you for the contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to). 

Also, since you are adding more return values, please update the `RETURN` documentation block to reflect that.

**Comment by Davidffry** [comment]
Hi @russoz, I complete the return section and I create the changelog fragment, tell me if something is wrong, first time for me.

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 1]
```diff
@@ -122,6 +122,226 @@
     type: str
     sample: ON
     version_added: 3.5.0
+
+server_name:
+    description: Server name
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:127]: The description should always have complete sentences.
```suggestion
    description: The server name.
```

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 2]
```diff
@@ -122,6 +122,226 @@
...
+    returned: always
+    type: str
+    sample: milo.company.com
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:130]:
```suggestion
    sample: milo.company.com
    version_added: 4.1.0
```
Same for the others.

File: changelogs/fragments/3701-hpilo_info_add_some_functionalities.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - hpilo_info - add other return values from HP ILO
```
Comment (felixfontein) [inline changelogs/fragments/3701-hpilo_info_add_some_functionalities.yml:3]:
```suggestion
  - hpilo_info - add return values ``server_name``, ... from HP ILO
```
(list all of them here)

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 4]
```diff
@@ -122,6 +122,226 @@
...
+pending_boot_mode:
+    description: Pending boot mode
+    returned: always
+    type: str
+    sample: UEFI
+
+persistent_boot:
+    description: System serial number
+    returned: always
+    type: list
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:283]:
```suggestion
    type: list
    elements: ... (fill type in here) ...
```

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 5]
```diff
@@ -122,6 +122,226 @@
...
+server_name:
+    description: Server name
+    returned: always
+    type: str
+    sample: milo.company.com
+
+snmp_im_settings:
+    description: System serial number
+    returned: always
+    type: dict
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:135]: If the format of the contents is known, you should add `contains:` and describe the fields in it as well.

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 6]
```diff
@@ -122,6 +122,226 @@
...
+    sample: normal
+
+pending_boot_mode:
+    description: Pending boot mode
+    returned: always
+    type: str
+    sample: UEFI
+
+persistent_boot:
+    description: System serial number
```
Comment (russoz) [inline plugins/modules/remote_management/hpilo/hpilo_info.py]: That description doesn't seem to match the returned value name

File: changelogs/fragments/3701-hpilo_info_add_some_functionalities.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - hpilo_info - add "server_name", "snmp_im_settings", "network_settings", "global_settings", "supported_boot_mode", "one_time_boot", "pending_boot_mode", "persistent_boot", "current_boot_mode", "security_msg", "all_licenses", "all_user_info", "all_users" return values from HP ILO
```
Comment (russoz) [inline changelogs/fragments/3701-hpilo_info_add_some_functionalities.yml:3]:
```suggestion
  - hpilo_info - add ``server_name``, ``snmp_im_settings``, ``network_settings``, ``global_settings``, ``supported_boot_mode``, ``one_time_boot``, ``pending_boot_mode``, ``persistent_boot``, ``current_boot_mode``, ``security_msg``, ``all_licenses``, ``all_user_info``, ``all_users`` return values from HP ILO.
```

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 8]
```diff
@@ -122,6 +122,239 @@
...
+    sample: UEFI
+    version_added: 4.1.0
+
+persistent_boot:
+    description: The persistant boot
+    returned: always
+    type: list
+    sample: 
+      persistent_boot:
+        -   - Boot0013
```
Comment (russoz) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:293]: is the double dashing intentional here?

**Comment by felixfontein** [comment]
@Davidffry just wanted to ask whether you are still working on this?

**Comment by Davidffry** [comment]
Yes 😅 I working on this

File: plugins/modules/remote_management/hpilo/hpilo_info.py (hunk 4 above)
Comment (felixfontein) [reply plugins/modules/remote_management/hpilo/hpilo_info.py:290]: Every list should say what type its elements are. Is this a list of strings, integers, dictionaries, ...?

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 9]
```diff
@@ -122,6 +122,239 @@
...
+    sample: 
+      security_msg:
+        security_msg: Disabled
+        security_msg_text: ''
+    version_added: 4.1.0
+
+all_licenses:
+    description: Display list of dict about the licences (expired or not)
+    returned: always
+    type: list of dict
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:327]:
```suggestion
    type: list
    elements: dict
```

**Comment by felixfontein** [comment]
@Davidffry ping

needs_info

**Pull request closed without merging**